test/lightning_web/live/project_live_test.exs

```diff
@@ -508,7 +508,14 @@ defmodule LightningWeb.ProjectLiveTest do
           ~p"/projects/#{project.id}/settings#vcs"
         )
 
-      Process.sleep(15)
+      Enum.take_while(0..30, fn _index ->
```

Comment:
I would have preferred something that could have looked at some underlying state to determine when to break out of the loop but (within the constraints of my limited Elixir) - it appears that an unsupervised Task runs and then passes the state to the Live view - so it does not look like there is something other than the rendered view that the test can reach into.